apis/contexts.go

```diff
@@ -198,3 +199,32 @@ func AllowDifferentNamespace(ctx context.Context) context.Context {
 func IsDifferentNamespaceAllowed(ctx context.Context) bool {
 	return ctx.Value(allowDifferentNamespace{}) != nil
 }
+
+// This is attached to contexts passed to webhook interfaces when the user has request DryRun mode.
+type isDryRun struct{}
+
+// WithDryRun is used to indicate that this call is in DryRun mode.
+func WithDryRun(ctx context.Context) context.Context {
+	return context.WithValue(ctx, isDryRun{}, struct{}{})
+}
+
+// IsDryRun indicates that this request is in DryRun mode.
+func IsDryRun(ctx context.Context) bool {
+	return ctx.Value(isDryRun{}) != nil
+}
+
+// Key to attach an operation verb to the context.
+type opVerb struct{}
+
+// WithOpVerb is used to attach an operation verb to the context.
+func WithOpVerb(ctx context.Context, op admissionv1beta1.Operation) context.Context {
```

Comment:
This package already has helpers for `IsInCreate`  & `IsInUpdate` - if we need additional operation checks we can add them in the same style.

Reply:
The problem with those interfaces is that in theory they can both be set! Plus you'd need to make a whole new type for every verb as they are supported/added.

For Callback to be a good generic interface let's try to simply pass the verb along for use. If we ever decide to move Validatable logic over to callback, we can switch these as well.

Reply:
Okay new idea. We can use IsDelete from context, but have a list of supported verbs for the Callback. Thoughts?

Reply:
sure